crates/payout_curve/src/lib.rs

```diff
@@ -0,0 +1,685 @@
+        initial_price,
+        liquidation_price_lower_bound
+            .to_u64()
+            .expect("to fit dec into u64"),
+        liquidation_price_upper_bound
+            .to_u64()
+            .expect("to fit dec into u64"),
+        &last_payout,
+        coordinator_direction,
+        quantity,
+    )?;
+
+    let (_, last_upper_range) = mid_range
+        .last()
+        .context("expect to have at least two elements in payout curve")?
+        .clone();
+
+    for (lower, upper) in mid_range {
+        pieces.push((lower, upper));
+    }
+    // if the upper bound is already [`BTCUSD_MAX_PRICE`] we don't have to add the upper bound
+    // anymore
+    if last_upper_range.event_outcome < BTCUSD_MAX_PRICE {
+        let upper_range_payout_points = calculate_upper_range_payouts(
+            coordinator_direction,
+            total_collateral,
+            last_upper_range,
+        )?;
+        pieces.push(upper_range_payout_points);
+    }
+
+    Ok(pieces)
+}
+
+/// Calculates the mid range payout points between lower liquidation and upper liquidation
+///
+/// Returns tuples of payout points, first item is lower point, next item is higher point of two
+/// points on the payout curve
+#[allow(clippy::too_many_arguments)]
+fn calculate_mid_range_payouts(
+    trader_collateral: u64,
+    coordinator_collateral: u64,
+    long_leverage: f32,
+    short_leverage: f32,
+    initial_price: Decimal,
+    lower_limit: u64,
+    upper_limit: u64,
+    last_payout: &PayoutPoint,
+    direction: Direction,
+    quantity: f32,
+) -> Result<Vec<(PayoutPoint, PayoutPoint)>> {
+    let total_collateral = trader_collateral + coordinator_collateral;
+
+    let pieces = (lower_limit..=upper_limit - 1)
+        .step_by(2)
+        .map(|current_price| {
+            let lower_event_outcome = current_price;
+            let lower_event_outcome_payout = if current_price == lower_limit {
+                last_payout.outcome_payout as i64
+            } else {
+                coordinator_collateral as i64
+                    + calculate_pnl(
+                        initial_price,
+                        Decimal::from(current_price),
+                        quantity,
+                        long_leverage.to_f32().context("to fit into f32")?,
+                        short_leverage.to_f32().context("to fit into f32")?,
+                        direction,
+                    )?
+            };
+            let lower_event_outcome_payout = if lower_event_outcome_payout <= 0 {
+                // we can't get negative payouts
+                0
+            } else if lower_event_outcome_payout as u64 > total_collateral {
+                // we can't payout more than there is in the contract, hence, we need to add this
+                // check
+                total_collateral
+            } else {
+                lower_event_outcome_payout as u64
+            };
+
+            let upper_event_outcome = current_price + 2;
```

Comment:
🔧 The step should be a variable or a constant.

Reply:
What do you mean?

Reply:
You've used a hard-coded 2 in a couple of places.

Reply:
ah, gotya